cmd/libsnap-confine-private/classic.c

```diff
@@ -1,15 +1,27 @@
 #include "config.h"
 #include "classic.h"
 
+#include <string.h>
+#include <stdio.h>
 #include <unistd.h>
 
+char *os_release = "/etc/os-release";
+
 bool is_running_on_classic_distribution()
 {
-	// NOTE: keep this list sorted please
-	return false
-	    || access("/var/lib/dpkg/status", F_OK) == 0
-	    || access("/var/lib/pacman", F_OK) == 0
-	    || access("/var/lib/portage", F_OK) == 0
-	    || access("/var/lib/rpm", F_OK) == 0
-	    || access("/sbin/procd", F_OK) == 0;
+	char buf[255];
+	int is_core = false;
+
+	FILE *f = fopen(os_release, "r");
+	if (f == NULL) {
+		return !is_core;
```

Comment:
I'd die here.

Reply:
I am not sure, it seems to be ok if a distro does not have a os-release file.We know its not core in this case because on core we have this file.